include/common/firmware/clocking.h

```diff
@@ -0,0 +1,11 @@
+#include "stm32g4xx_hal_conf.h"
```

Comment:
Is this include needed?

Reply:
I can either include it in this header file or the `.c` file. We need the hal_conf because of the HAL functions we are using in both `errors.c` and `clocking.c`.

Reply:
I understand. 

I always try to limit includes to `c` files whenever possible. Purely to reduce compilation times. If a header file changes then all files that include that file will need to be rebuilt.  That can have a cascading effect.

Reply:
Need an include guard (#pragma once)

Reply:
include guard comment in the wrong place?

I agree with amit, for a couple other reasons. In general, having include files that include others that they do not _directly_ require to function bloats compilation time but also hides bugs where some .c(pp) file doesn't include something but does include this, and it gets papered over.

Having specifically the hal files included here also makes this basically not usable from c++ because the hal does stuff that isn't really good to do in c++ (volatile increment-assignment). We should avoid it if possible.

Reply:
I see. That makes sense, will definitely change it.